src/js/storage.js

```diff
@@ -244,6 +252,9 @@ BadgerPen.prototype = {
    */
   setupHeuristicAction: function(domain, action){
     this._setupDomainAction(domain, action, "heuristicAction");
+    if (this.isActionMapEmptyForDomain(domain)) {
+      this.getBadgerStorageObject('action_map').deleteItem(domain);
+    }
```

Comment:
Is this `if` condition ever true? A quick check in the code doesn't reveal any cases where `setupHeuristicAction` is called with either `""` or `constants.NO_TRACKING`.